x-pack/plugins/observability_solution/apm/server/utils/es_fields_mappings/es_fields_mappings_helpers.ts

```diff
@@ -0,0 +1,16 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import type { Fields } from './types';
+
+export const normalizeValue = <T>(field: unknown[] | unknown): T => {
+  return (Array.isArray(field) && field.length === 1 ? field[0] : field) as T;
+};
+
+export const isOptionalFieldDefined = (fields: Fields, fieldToCheck: string): boolean =>
+  Object.keys(fields)
+    .filter((field) => field.includes(fieldToCheck))
+    .some((fieldName) => normalizeValue(fields[fieldName]) !== undefined);
```

Comment:
why is this needed? I assume if the key is present that's enough?

Reply:
My concern is that it can still be `undefined` but I am not sure if it's a valid one 🤔 So if it's undefined it shouldn't be returned, right?

Reply:
exactly, it won't be there